### PR TITLE
Added possibility to profile any command by setting the option --xhprof

### PR DESCRIPTION
**Pull request by tsteur**

### CODE REVIEW

File: core/Console.php [hunk 1]
```diff
@@ -34,11 +34,23 @@ public function __construct()
 
         $this->getDefinition()->addOption($option);
 
+        $option = new InputOption('xhprof',
+            null,
+            InputOption::VALUE_NONE,
+            'Enable profiling with XHProf'
+        );
+
+        $this->getDefinition()->addOption($option);
```
Comment (diosmosis) [inline core/Console.php]: Can you add this only if development mode is enabled?

**Comment by diosmosis** [comment]
Can we remove the xhprof param from the core:archive command?

File: core/Console.php (hunk 1 above)
Comment (tsteur) [inline core/Console.php]: Nah, you don't want to profile when development mode is enabled... it wouldn't cache, perform extra checks etc.

**Comment by tsteur** [comment]
The core:archive does some additional work, eg it does forward the `--xhprof` param to all requests but we can remove the option itself.

File: core/Console.php (hunk 1 above)
Comment (diosmosis) [inline core/Console.php]: In that case, can we echo a warning or throw an exception when trying to profile w/ development mode is enabled (would do this when calling setupProfilerXhprof for the main run)?

Comment (tsteur) [inline core/Console.php]: Not sure, we could log a warning and remove it once it becomes a problem. Eg someone might enable development mode on purpose to profile things uncached etc. Adding this logger would be a new issue I reckon as we would need to do this also when using `xhprof=1` in a URL.

Comment (diosmosis) [inline core/Console.php]: Maybe in the summary output (that includes a link to the profile) we can print whether Development mode is enabled, and a note that says nothing is cached and you may want to run w/ it disabled, etc.

Comment (tsteur) [inline core/Console.php]: Created https://github.com/piwik/piwik/issues/7708

**Comment by diosmosis** [comment]
@tsteur If you rebase/merge, I'll merge the PR.

**Pull request merged**